frontend/src/components/pages/HomePage/LogRecordsTable.tsx

```diff
@@ -8,66 +8,243 @@ import {
   Th,
   Thead,
   Tr,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  IconButton,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
 } from "@chakra-ui/react";
+import { VscKebabVertical } from "react-icons/vsc";
+
 import { LogRecord } from "../../../types/LogRecordTypes";
 import getFormattedDateAndTime from "../../../utils/DateUtils";
 
+import AuthContext from "../../../contexts/AuthContext";
+
+import DeleteConfirmation from "../../common/DeleteConfirmation";
+import EditResident from "../../forms/EditLog";
+import LogRecordAPIClient from "../../../APIClients/LogRecordAPIClient";
+import ResidentAPIClient from "../../../APIClients/ResidentAPIClient";
+import UserAPIClient from "../../../APIClients/UserAPIClient";
+import { UserLabel } from "../../../types/UserTypes";
+
 type Props = {
   logRecords: LogRecord[];
   tableRef: RefObject<HTMLDivElement>;
+  getRecords: (page_number: number) => Promise<void>;
+  countRecords: () => Promise<void>;
+  setUserPageNum: React.Dispatch<React.SetStateAction<number>>;
 };
 
 const LogRecordsTable = ({
   logRecords,
   tableRef,
+  getRecords,
+  countRecords,
+  setUserPageNum,
 }: Props): React.ReactElement => {
+  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
+
+  const [showAlert, setShowAlert] = useState(false);
+
+  // Menu states
+  const [deleteOpenMap, setDeleteOpenMap] = useState<{
+    [key: number]: boolean;
+  }>({});
+  const [editOpenMap, setEditOpenMap] = useState<{ [key: number]: boolean }>(
+    {},
+  );
+
+  // Dropdown option states
+  const [employeeOptions, setEmployeeOptions] = useState<UserLabel[]>([]);
+  const [residentOptions, setResidentOptions] = useState<UserLabel[]>([]);
+
+  // Handle delete confirmation toggle
+  const handleDeleteToggle = (logId: number) => {
```

Comment:
I've never seen this before, curious how it works (seems like we're just toggling logId?)

EDIT: Nevermind I see what this is doing - it's just keeping a map of what modal's are toggled open.